gcloud-java-dns/src/main/java/com/google/gcloud/dns/Zone.java

```diff
@@ -0,0 +1,256 @@
+    }
+    if (updated == null && zoneInfo.name() != null) {
+      // zone was not found by id or id is not set at all
+      updated = dns.applyChangeRequest(changeRequest, zoneInfo.name(), options);
+    }
+    return updated;
+  }
+
+  /**
+   * Retrieves an updated information about a change request previously submitted to be applied to
+   * this zone. First searches for the zone by ID and if not found then by name. Returns a {@link
+   * ChangeRequest} if found and null is the zone or the change request was not found.
+   *
+   * @param options Optional restriction on what fields of {@link ChangeRequest} should be returned
+   * by the service
+   * @return ChangeRequest with updated information of null if the change request or zone was not
+   * found.
+   * @throws DnsException upon failure
+   * @throws NullPointerException if both zone ID and name are not initialized
+   * @throws NullPointerException if the change request does not have initialized id
+   */
+  public ChangeRequest getChangeRequest(ChangeRequest changeRequest,
+                                        Dns.ChangeRequestOption... options) {
+    checkNameOrIdNotNull();
+    checkNotNull(changeRequest);
+    checkNotNull(changeRequest.id());
+    ChangeRequest updated = null;
+    if (zoneInfo.id() != null) {
+      updated = dns.getChangeRequest(changeRequest, zoneInfo.id(), options);
+    }
+    if (updated == null && zoneInfo.name() != null) {
+      // zone was not found by id or id is not set at all
+      updated = dns.getChangeRequest(changeRequest, zoneInfo.name(), options);
+    }
+    return updated;
+  }
+
+  /**
+   * Retrieves all change requests for this zone. First searches for the zone by ID and if not found
+   * then by name. Returns a page of {@link ChangeRequest}s or null if the zone is not found.
+   *
+   * @param options Optional restriction on listing and on what fields of {@link ChangeRequest}s
+   * should be returned by the service
+   * @return {@code Page<ChangeRequest>}, a page of change requests, or null if the zone is not
+   * found
+   * @throws DnsException upon failure
+   * @throws NullPointerException if both zone ID and name are not initialized
+   */
+  public Page<ChangeRequest> listChangeRequests(Dns.ChangeRequestListOption... options) {
+    checkNameOrIdNotNull();
+    Page<ChangeRequest> changeRequests = null;
+    if (zoneInfo.id() != null) {
+      changeRequests = dns.listChangeRequests(zoneInfo.id(), options);
+    }
+    if (changeRequests == null && zoneInfo.name() != null) {
+      // zone was not found by id or id is not set at all
+      changeRequests = dns.listChangeRequests(zoneInfo.name(), options);
+    }
+    return changeRequests;
+  }
+
+  /**
+   * Check that at least one of name and ID are initialized and throw and exception if not.
+   */
+  private void checkNameOrIdNotNull() {
+    if (zoneInfo != null && zoneInfo.id() == null && zoneInfo.name() == null) {
```

Comment:
Nit: also [checkArgument](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Preconditions.html#checkArgument(boolean,%20java.lang.Object)) could be used